GitUI/Editor/FileViewer.cs

```diff
@@ -1557,12 +1548,20 @@ private void ApplySelectedLines(bool allFile, bool reverse)
                 "--whitespace=nowarn"
             };
 
-            string output = Module.GitExecutable.GetOutput(args, patch, throwOnErrorOutput: false);
-            ProcessApplyOutput(output, patch);
+            ProcessApplyOutput(args, patch);
         }
 
-        private void ProcessApplyOutput(string output, byte[] patch, bool patchUpdateDiff = false)
+        private void ProcessApplyOutput(GitArgumentBuilder args, byte[] patch, bool patchUpdateDiff = false)
         {
+            ExecutionResult result = Module.GitExecutable.Execute(args, inputWriter => inputWriter.BaseStream.Write(patch), throwOnErrorOutput: false);
+            string output = result.AllOutput.Trim();
+            if (EnvUtils.RunningOnWindows())
+            {
+                // remove file mode warnings
+                Regex regEx = new("warning: .*has type .* expected .*");
+                output = output.RemoveLines(regEx.IsMatch);
+            }
```

Comment:
I'm confused about this one... Warnings would be in the error stream, why not ignore those completely?

Reply:
AllOutput is stdout+stderr

Reply:
That's my underlying question - is stderr used for anything useful? Previously the output would contain both of stdout and stderr, so we needed this filtering. Is it still required with this change?

Reply:
In this case, this is what we want to possibly display to the user.

See GitModule.createErrorGitItemStatus() where an "error item" is created from standard error at failures.

In #9468 stderr is used in popups at failures in the same way. In that PR I am not sure if this is too aggressive or not, so Trace will print a few more printouts (such printouts should be logged some way).

This "clean" could be lazy but the processing is a fraction of the time to patch files so it will not be noticeable.